client/src/main/scala/io/delta/sharing/client/DeltaSharingProfileProvider.scala

```diff
@@ -17,22 +17,138 @@
 package io.delta.sharing.client
 
 import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Locale
 
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.{DeserializationContext, JsonDeserializer}
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.node.ObjectNode
 import org.apache.commons.io.IOUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.spark.delta.sharing.TableRefreshResult
 
+import io.delta.sharing.client.DeltaSharingProfile.validateNotNullAndEmpty
 import io.delta.sharing.client.util.JsonUtils
 
-case class DeltaSharingProfile(
-    shareCredentialsVersion: Option[Int],
-    endpoint: String = null,
+@JsonDeserialize(using = classOf[DeltaSharingProfileDeserializer])
+sealed trait DeltaSharingProfile {
+  val shareCredentialsVersion: Option[Int]
+  val endpoint: String
+  val `type` : String
```

Comment:
call this profileType (to avoid backticks)?

Reply:
sure, renamed.